pkg/types/v1/config.go

```diff
@@ -220,25 +220,11 @@ func (r *EmptySpec) Sanitize() error {
 func (r *EmptySpec) ShouldReboot() bool   { return false }
 func (r *EmptySpec) ShouldShutdown() bool { return false }
 
-// Partition struct represents a partition with its commonly configurable values, size in MiB
-type Partition struct {
-	Name            string   `yaml:"-"`
-	FilesystemLabel string   `yaml:"label,omitempty" mapstructure:"label"`
-	Size            uint     `yaml:"size,omitempty" mapstructure:"size"`
-	FS              string   `yaml:"fs,omitempty" mapstrcuture:"fs"`
-	Flags           []string `yaml:"flags,omitempty" mapstrcuture:"flags"`
-	MountPoint      string   `yaml:"-"`
-	Path            string   `yaml:"-"`
-	Disk            string   `yaml:"-"`
-}
-
-type PartitionList []*Partition
-
 // GetByName gets a partitions by its name from the PartitionList
-func (pl PartitionList) GetByName(name string) *Partition {
-	var part *Partition
+func GetByName(name string, partitionList types.PartitionList) *types.Partition {
```

Comment:
The name was kept generic because the receiver was a PartitionList. Now that it doesn't have a receiver a call to this will simply look like `GetByName` which is very generic and doesn't reveal anything about what is there to "Get".

Maybe `GetPartitionByName` ?

Alternatively, can we keep the receiver (using the new type)?

Reply:
same this below in `GetByLabel`

Reply:
ah yes, good catch, I forgot about this tbh, it should probably also be moved to utils directly, makes no sense to have it here. Thanks for the reminder!

Reply:
did merge both into a `getPartitionByNameOrLabel` function instead. Keeping it in the same file as its only use there